src/nano-network-api.js

```diff
@@ -370,8 +418,28 @@ export class NanoNetworkApi {
         return true;
     }
 
+    _bindEvents() {
+        this._consensus.on('syncing', e => this._onConsensusSyncing());
+        this._consensus.on('established', e => this.__consensusEstablished());
+        this._consensus.on('lost', e => this._consensusLost());
+
+        this._consensus.on('transaction-relayed', tx => this._transactionRelayed(tx));
+
+        // this._consensus.on('sync-finished', e => console.log('consensus sync-finished'));
+        // this._consensus.on('sync-failed', e => console.log('consensus sync-failed'));
+        // this._consensus.on('sync-chain-proof', e => console.log('consensus sync-chain-proof'));
+        // this._consensus.on('verify-chain-proof', e => console.log('consensus verify-chain-proof'));
+
+        this._consensus.blockchain.on('head-changed', block => this._headChanged(block.header));
+        this._consensus.mempool.on('transaction-added', tx => this._transactionAdded(tx));
+        this._consensus.mempool.on('transaction-expired', tx => this._transactionExpired(tx));
+        this._consensus.mempool.on('transaction-mined', (tx, header) => this._transactionMined(tx, header));
+        this._consensus.network.on('peers-changed', () => this._onPeersChanged());
+    }
+
     async _headChanged(header) {
-        if (!this._consensus.established) return;
+        if (!this._consensus.established || this._knownHead && this._knownHead.equals(header)) return;
```

Comment:
Is this condition working correctly? I think OR is evaluated before AND, but I'm not 100% sure. Brackets around the AND condition would help remove any doubts here.
```suggestion
        if (!this._consensus.established || (this._knownHead && this._knownHead.equals(header))) return;
```

Reply:
`&&` has a higher priority but I added parentheses to make it clear